Dockerfile

```diff
@@ -55,59 +50,94 @@ RUN if test -n "$FLUTTER_BRANCH" -a "$FLUTTER_BRANCH" != "stable" ; then \
     fi
 
 # Set up Flutter
+# NOTE You will get a warning "Woah! You appear to be trying to run flutter as root."  
+# and this is to be disregarded since this image is never deployed to production.
+# It is not worth setting up a Docker user or mirroring host, but you may run into 
+# conflicts with the local host repo and the Docker one.
 RUN flutter doctor
 RUN flutter --version
 RUN dart pub get
 
+
+
+# ============== NODEJS INTSALL ==============
+# TODO recommended to move to a SHA based install for security reasons
+FROM flutter AS node
+
+RUN curl -sL https://deb.nodesource.com/setup_17.x -o node_setup.sh && \
+      bash node_setup.sh 1> /dev/null
+RUN apt-get update -q && apt-get install -yq --no-install-recommends \
+      nodejs \
+    && rm -rf /var/lib/apt/lists/*
+# Ensure latest NPM, install global Firebase CLI
+RUN npm install -g npm firebase-tools@11.0.1
```

Comment:
It appears this change has broken the `check_link_references` in my pull request? `Node 17` seems to not be supported by `firebase-tools@11.0.1` (if I'm reading the error right)?

@btobin, can you take a look at: https://github.com/flutter/website/pull/7740 ? Thanks!